src/NorthwindWebsite/Presentation/Controllers/CategoriesController.cs

```diff
@@ -20,4 +26,64 @@ public async Task<IActionResult> Index()
 
         return View("Index", categories);
     }
+
+    [HttpGet]
+    public async Task<IActionResult> ImageUpload(int id)
+    {
+        var fileUploadModel = await _categoryService.GetFileUploadModel(id);
+
+        return View("Upload", fileUploadModel);
+    }
+
+    [HttpPost]
+    [ValidateAntiForgeryToken]
+    public async Task<IActionResult> ImageUpload(FileUploadDto fileUploadModel)
+    {
+        fileUploadModel.MaximumFileSize = _appSettings.FileUploadOptions.ImageMaxSize;
+
+        if (!ModelState.IsValid)
+        {
+            ModelState.AddModelError(string.Empty, "Please select a file.");
+            return View("Upload", fileUploadModel);
+        }
+
+        var permittedExtensions = _appSettings.FileUploadOptions.ImageFileFormats
+            .Split(", ").ToArray();
+
+        var uploadedFileExtension = fileUploadModel.FileUpload.ContentType.Split("/")[1];
```

Comment:
Let's move it to separate util function and specify in its name the purpose of `Split("/")[1]`